### PR TITLE
2171: Staff lists

### PR DESCRIPTION
**Pull request by BirkAndMe**
http://platform.dandigbib.org/issues/2171

### CODE REVIEW

File: modules/p2/ding_list/theme/theme.inc [hunk 1]
```diff
@@ -164,24 +165,28 @@ function theme_ding_list_lists($variables) {
       }
     }
 
+    if (!empty($rows)) {
+      $out .= theme('item_list', array(
```
Comment (JrgenGNielsen) [inline modules/p2/ding_list/theme/theme.inc]: Avoid calling theme() directly. It is preferable to replace direct calls to the theme() function with calls to drupal_render() by passing a render array with a #theme key to drupal_render(), which in turn calls theme().
See: https://api.drupal.org/api/drupal/includes!theme.inc/function/theme/7.x

File: modules/p2/ding_list/theme/theme.inc [hunk 2]
```diff
@@ -164,24 +165,28 @@ function theme_ding_list_lists($variables) {
...
+        'title' => $group['title'],
+        'type' => 'ul',
+        'attributes' => array(),
+      ));
+    }
+  }
+
+  // Footer links.
+  if ($variables['uid'] == $user->uid) {
     $out .= theme('item_list', array(
```
Comment (JrgenGNielsen) [inline modules/p2/ding_list/theme/theme.inc]: ditto

**Comment by JrgenGNielsen** [comment]
Jeg har en kommentar vedr. brug af theme()

File: modules/p2/ding_list/theme/theme.inc [hunk 3]
```diff
@@ -164,24 +165,33 @@ function theme_ding_list_lists($variables) {
...
+    if (!empty($rows)) {
+      $render = array(
+        '#theme' => 'item_list',
+        '#title' => $group['title'],
+        '#items' => $rows,
+        '#title' => $group['title'],
+        '#type' => 'ul',
+        '#attributes' => array(),
+      );
+      $out .= drupal_render($render);
```
Comment (cableman) [inline modules/p2/ding_list/theme/theme.inc]: Calling drupal_render is basically the same as theme().

You theme functions should return a render array and let the theme layer do the rendering. So the front-end has a change to alter in style and content.

File: modules/p2/ding_list/theme/theme.inc [hunk 4]
```diff
@@ -164,24 +165,33 @@ function theme_ding_list_lists($variables) {
...
+      '#theme' => 'item_list',
+      '#items' => array(
+        l(t("Mark all notifications as read"), "ding_message/purge_user_messages", array('attributes' => array('class' => array('button')))),
+        l(t("Create new list"), "list/create", array('attributes' => array('class' => array('button')))),
+      ),
+      '#title' => "",
+      '#type' => 'ul',
+      '#attributes' => array('class' => array('bottom-links')),
+    );
+    $out .= drupal_render($render);
```
Comment (cableman) [inline modules/p2/ding_list/theme/theme.inc]: Calling drupal_render is basically the same as theme().

You theme functions should return a render array and let the theme layer do the rendering. So the front-end has a change to alter in style and content.

File: modules/p2/ding_list/theme/theme.inc (hunk 3 above)
Comment (BirkAndMe) [inline modules/p2/ding_list/theme/theme.inc]: It's called from inside a theme_HOOK() function, we could move the creation of the item_list to a preprocessor, but we'd still need to call drupal_render inside the theme_HOOK().

File: modules/p2/ding_list/theme/theme.inc (hunk 4 above)
Comment (BirkAndMe) [inline modules/p2/ding_list/theme/theme.inc]: It's called from inside a theme_HOOK() function, we could move the creation of the item_list to a preprocessor, but we'd still need to call drupal_render inside the theme_HOOK().

**Pull request merged**